src/components/overlays/delete-event.overlay.tsx

```diff
@@ -0,0 +1,129 @@
+import React, { useEffect, useState } from "react"
+import OverlayContainer from "./overlay-container"
+import { getOverlay } from "@/helpers/overlay.helper"
+import { ClockIcon, MapPinIcon, XMarkIcon } from "@heroicons/react/24/outline"
+import { useSupabaseClient } from "@supabase/auth-helpers-react"
+import { Database } from "@/types/database"
+import { Event } from "@/types/database-types"
+import { getEventDate, getEventTime, getEventTitle } from "@/helpers/event.helper"
+import { CalendarIcon } from "@heroicons/react/24/solid"
+import {deleteEvent} from "@/services/event.service";
+import {useRouter} from "next/router";
+import {toast} from "react-toastify";
+
+interface DeleteEventOverlayProps {
+  onClose: (showOverlay: boolean) => void
+  event: Event
+}
+
+const DeleteEventOverlay = ({ onClose, event }: DeleteEventOverlayProps) => {
+  const supabase = useSupabaseClient<Database>()
+  const [overlayIsOpen, setOverlayIsOpen] = useState<boolean>(false)
+  const [showInput, setShowInput] = useState(false)
+  const [deleteButtonIsActive, setDeleteButtonIsActive] = useState(false)
+  const router = useRouter();
+  useEffect(() => setOverlayIsOpen(true), [])
+
+  const waitForTransition = (isOpen: boolean) => {
+    setOverlayIsOpen(false)
+    setTimeout(() => onClose(isOpen), 300)
+  }
+
+  const handleChange = (value: string) => {
+    value == getEventTitle(event.event_type)
+      ? setDeleteButtonIsActive(true)
+      : setDeleteButtonIsActive(false)
+  }
+
+  const removeEvent = () => {
+    const confirmed = window.confirm("Are you sure you want to sign out?")
```

Comment:
what do you mean by sign out? of the app or the event? or do you mean remove event?

Reply:
copypaste